### ISSUE TITLE
[Feature request] Desktop messaging

### ISSUE THREAD

**Issue by KaKi87** [octoshrimpy/quik#9]
> Being able to text from your browser
— moezbhatti/qksms#1787

Thanks

**Comment by octoshrimpy** [comment]
this will require sms syncing with your own DB, probably using pub/priv keys (a la ssh) for encryption/decryption.

**Comment by octoshrimpy** [comment]
putting more thought into this over the next week.

**Comment by s-light** [comment]
Have a look at kde connect:
https://f-droid.org/en/packages/org.kde.kdeconnect_tp/
There us a SMS desktop app..
it is very simple. But basics are working...

**Comment by KaKi87** [comment]
> very simple

Too much simple.

No access to anything dated before firt sync, no search, no deletion, no receipt indicator...

**Comment by roo79x** [comment]
This might not suit everyone's needs exactly. But as an alternative or even an interim solution there is KDE Connect https://kdeconnect.kde.org/
It has the ability to have sms on pc. It is available for many OS's as well.
https://kdeconnect.kde.org/download.html

**Comment by roo79x** [comment]
> Have a look at kde connect: https://f-droid.org/en/packages/org.kde.kdeconnect_tp/ There us a SMS desktop app.. it is very simple. But basics are working...

I Have used this for years. It's excellent.

**Comment by roo79x** [comment]
> > very simple
> 
> Too much simple.
> 
> No access to anything dated before first sync, no search, no deletion, no receipt indicator...

Can you point to an SMS / MMS (Not RCS like google messages etc) That can do those things please? as I'm yet to find any that can do all those things while not giving away your data to some big corporate or shady entity. I don't know about Windows or Mac variants of KDE Connect but on Linux KDE Connect has search, It shows you a notification when a new message arrives. You are right that there is no deletion on Desktop. I can't comment on receipt indicator as I don't use that feature.

**Comment by KaKi87** [comment]
> Can you point to an SMS / MMS (Not RCS like google messages etc) That can do those things please? as I'm yet to find any that can do all those things while not giving away your data to some big corporate or shady entity.

If there were, I would be using it, instead of asking here. 😅

> I don't know about Windows or Mac variants of KDE Connect

Neither do I.

> Linux KDE Connect has search

Search for conversations yes (and even then, only loaded ones), but no search for messages inside conversations.

>

**Comment by KaKi87** [comment]
PS : Google doesn't require enabling RCS to send messages on desktop, SMS/MMS are supported, since a while before RCS was deployed.

**Comment by roo79x** [comment]
> PS : Google doesn't require enabling RCS to send messages on desktop, SMS/MMS are supported, since a while before RCS was deployed.

That is right. But the messages for web uses that protocol as far as I'm aware even if you just use standard sms/mms. On the server end it converts them. If I can remember where I seen the article about it I'll post it. I am not knowledgeable on all things sms/mms. But I do remember reading that is how google does it.

**Comment by Zulgrib** [comment]
Adding a web server to the sms application would severely increase the attack surface.

**Comment by KaKi87** [comment]
Nobody ever said it had to be implemented that way.
We just need the feature.

**Comment by octoshrimpy** [comment]
> **Is your feature request related to a problem? If so, please describe the problem.**
Not really a "problem", but lack of really important functionality - a web messaging allows to "copy" data from/into SMS which is very convenient, also typing SMS with PC keyboard is way more effective. Would be nice to have something what Google provides.
>
> **Describe the feature you'd like**
Exact behavioral copy of google web messages. A messaging app in JS possibly that connects to the phone via WiFi or LAN (best scenario, if that is harder to achieve, may be any network, but I suppose it would be even nice to specify the local/lan phone IP if possible or something, anything will do, just let it happen :D )
>
> **Additional context**
You know where to find google web messages. https://messages.google.com/web/authentication
>
> If this will be only a stub that uses LAN then there will be no problem with hosting, privacy and so on.
> PS. I will be happy to become a tester (im also a dev, so it may help)
> - @KaJakKrzysztof

**Comment by KaJakKrzysztof** [comment]
I'm sorry to disagree, but #9 is not even close, the simple proof is that in the first comment you already proposed a database syncing (which would probably be ok for a desktop app), but here is rather overhead.

Pleas do not mix topics. Web app is a web app, it can be used on the phone, on tablet, on raspberry pi, on anything. Desktop app is system-binded (ie. windows exe or linux elf) and is completely different story

**Comment by KaJakKrzysztof** [comment]
By the way, web app does not need to be even written by you, you just need to provide an API.

**Comment by KaJakKrzysztof** [comment]
> I feel that the way I think it would be best to go about this is to rewrite the Android app in Jetpack Compose and then convert it to Compose Multiplatform which will allow desktop and web interfaces to be built easily, with exactly the same UI. Just my two cents.

Exactly how it would solve the web messaging? This is not about building app (messaging app), but only a "screen-keyboard" API. Nothing more, the app itself HAS TO be on the phone (ie. sim card will not be magically teleported wherever you go with the web app).

The idea is for entirely different feature request. And my few cents - its complete waste of time and of no use to anyone using QUIK/QKSMS (as we all want just native android app with the web api)
